src/app/profile/view.nim

```diff
@@ -195,11 +195,18 @@ QtObject:
     self.mutedChatsListChanged()
     self.mutedContactsListChanged()
 
+  proc contactsChanged*(self: ProfileView) {.signal.}
+
+  proc addContact*(self: ProfileView, pubKey: string): string {.slot.} =
+    result = self.contacts.addContact(pubKey)
+    self.contactsChanged()
```

Comment:
Did you add this because the list wasn't updated when calling `addContact` directly on `contacts`? If so, maybe there was a signal missing in `contacts`

Reply:
Yes exactly I've added it here because the list won't update otherwise. I wasn't aware that we can emit a signal on an abstract list view itself? Are we already doing that somewhere else in the app?

Reply:
I checked how `addContact` works and I'm not sure exactly how the new contact is added to the list. I assume it's a signal that status-go sends that we process and then we add it to `contact_list`. Anyway, if it does, then it should call the signal anyway, since it has `beginResetModel`. Another way maybe would be to call `addContactToList` with the new contact, but I don't think it would work since we don't have all the data (it comes from status-go).

Anyway, I'm not sure